sklearn/preprocessing/_discretization.py

```diff
@@ -250,6 +260,11 @@ def fit(self, X, y=None):
         n_features = X.shape[1]
         n_bins = self._validate_n_bins(n_features)
 
+        if sample_weight is not None:
+            weights = np.array(
```

Comment:
If I replace the variable `weights` by `sample_weight` with `copy=True`, the `sample_weights` passed in parameter might be overwritten and have its value changed. I used variable `weights` to avoid this.

Reply:
Yes but this is fine because in the rest of the code, we always want to use the validated `sample_weight`.

Reply:
Change `weights` to `sample_weight`